x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -2939,4 +2946,30 @@ public void testNamedFunctionArgumentWithUnsupportedNamedParameterTypes() {
             );
         }
     }
+
+    public void testValidJoinPattern() {
+        var basePattern = randomIndexPatterns();
+        var joinPattern = randomIndexPattern(without(WILDCARD_PATTERN));
+        var onField = randomIdentifier();
+        var type = randomFrom("", "LOOKUP ");
+
```

Comment:
The empty string looks wrong - `FROM idx | JOIN other_idx ON field` is not supported, but by the test below, we parse this as a lookup join o.O

Consider removing the plain join, but it's also fine as-is due to this only testing the parsing itself.